### PR TITLE
Introduce noms version 8. Use it to guard type simplification.

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

**Comment by aboodman** [comment]
Main change is in version.go (factored out from constants.go). Most of the remaining changes are fallout.

One notable exception to above is that I had to dork around with util/command.go and friends to get this flag to show up in all the noms subcommands without having to modify them all individually.

**Comment by aboodman** [comment]
Working on js... will add to this patch.

File: js/noms/src/version.js [hunk 1]
```diff
@@ -4,4 +4,24 @@
 
 // @flow
 
-export default '7';
+const stable = '7';
+const next = '8';
+let useNext = false;
+
+export default {
```
Comment (arv) [inline js/noms/src/version.js]: This is an anti pattern. Export all the functions instead. You might still want to `export default '7';`

**Comment by arv** [comment]
LGTM

**Pull request merged**